go/vt/vtgate/planbuilder/operators/queryprojection.go

```diff
@@ -309,6 +310,19 @@ func (qp *QueryProjection) addSelectExpressions(sel *sqlparser.Select) error {
 	return nil
 }
 
+func containsAggr(e sqlparser.SQLNode) (containsAggr bool) {
+	_ = sqlparser.Walk(func(node sqlparser.SQLNode) (bool, error) {
+		if _, isAggr := node.(sqlparser.AggrFunc); isAggr {
+			containsAggr = true
+			return false, io.EOF
+		}
+
+		_, isSubquery := node.(*sqlparser.Subquery)
+		return !isSubquery, nil
+	}, e)
+	return
+}
+
```

Comment:
There is `containsAggregation` in `SQL_Builder.go` as well. They are similar apart from how they handle Offsets. Maybe they can be merged into just one function.